apps/tasks/views.py

```diff
@@ -1,36 +1,73 @@
-from rest_framework import status, filters
+import http
+
+from django.conf import settings
+
+from rest_framework import filters
 from rest_framework.decorators import action
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 from rest_framework.viewsets import ModelViewSet
 
-from apps.tasks.models import Task
-from apps.tasks.serializers import TaskSerializer, TaskAssignSerializer, DisplayTaskSerializer
+from apps.common.helpers import send_user_email
+from apps.tasks.models import Task, Comment
+from apps.tasks.serializers import TaskSerializer, TaskAssignSerializer, TaskCommentSerializer
 
 
 class TaskViewSet(ModelViewSet):
     serializer_class = TaskSerializer
     queryset = Task.objects.all()
     permission_classes = (IsAuthenticated,)
     filter_backends = [filters.SearchFilter]
-    search_fields = ['=id']
+    search_fields = ['$title']
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+    @action(detail=False, methods=['PUT'], serializer_class=TaskAssignSerializer)
+    def assign(self, request):
+        Task.objects.filter(pk=self.request.data['id']).update(assigned_to=self.request.data['assigned_to'])
```

Comment:
Mai bine sa nu folosesti `request.data`, ea nu e validata. Mai bine sa folosesti `serializer.validated_data`. In genral in cazul la acest endpoint poti sa folosesti path parameters. Poti sa setezi `detail=True` in action si atunci nu o sa ai nevoie de `request.data['id'] ca o sa-l primesti ca parametru functiei` si poti sa setezi `url_path` cu parametru in loc de `self.request.data['assigned_to']`.

Uite [aici](https://www.django-rest-framework.org/api-guide/viewsets/#routing-additional-http-methods-for-extra-actions) exemplu cu `detail=True` si [aici](https://stackoverflow.com/a/70098847) exemplu cu URL path parameter.

Reply:
Done